fab3/types/types.go

```diff
@@ -200,6 +200,17 @@ type Log struct {
 	Index       string   `json:"logIndex"`
 }
 
+// MarshalJSON will json marshal the transaction receipt and set Logs to an empty []Log if none returned
+func (txReceipt *TxReceipt) MarshalJSON() ([]byte, error) {
+	receipt := *txReceipt
+
+	if receipt.Logs == nil || len(receipt.Logs) == 0 {
+		receipt.Logs = []Log{}
+	}
```

Comment:
Sorry, what does the check for 0 length help us with?

Reply:
Was making sure all bases are covered, but thinking about it now, this may not be needed actually.

Reply:
My analysis is if we have the nil slice come in, it will fail both of those checks, so the only check that would catch the second clause (and make some difference in output) is for an empty slice, that has more than 0 capacity. Even so, it wouldn't change the resulting output.

I think we can drop the length check.

Reply:
My analysis is if we have the nil slice come in, it will fail both of those checks, so the only check that would catch the second clause (and make some difference in output) is for an empty slice, that has more than 0 capacity. Even so, it wouldn't change the resulting output.

I think we can drop the length check.